app/controllers/waste_exemptions_engine/forms_controller.rb

```diff
@@ -67,12 +67,9 @@ def setup_checks_pass?
     def set_workflow_state
       return unless state_can_navigate_flexibly?(@transient_registration.workflow_state)
       return unless state_can_navigate_flexibly?(requested_state)
+      return unless @transient_registration.persisted?
 
-      if @transient_registration.persisted?
-        update_workflow_state(requested_state)
-      else
-        assign_workflow_state(requested_state)
```

Comment:
Does this mean we can delete `def assign_workflow_state`?

Reply:
Oh yeah we can!